core/controllers/reader.py

```diff
@@ -1274,3 +1299,103 @@ def put(self, entity_type, entity_id):
             entity_type, state_reference,
             interaction_id, answer, answer_details)
         self.render_json({})
+
+
+class CheckpointCompletedEventHandler(base.BaseHandler):
```

Comment:
@IamEzio We talked updating latest_visited_checkpoint_state_name in the same post request rather than calling it separately from frontend when a checkpoint is completed today.  I think that needs to be incorporated.